File: edupilot/lib/models/coupon/coupon.dart
import 'package:flutter/material.dart';

class Coupon{
  Coupon({
    required this.id,
    required this.icon,
    required this.name,
    required this.description,
    required this.fee,
  });

  final String id;
  final IconData? icon;
  final String name;
  final String description;
  final int fee;
}

File: edupilot_api/EduPilot.Api/Data/Models/Coupon.cs
namespace EduPilot.Api.Data.Models
{
    public class Coupon
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Icon { get; set; }
        public int Fee { get; set; }
        public bool IsAvailable { get; set; }
        public List<ClaimedCoupon> ClaimedCoupons { get; set; }
    }
}